provider/provider.go

```diff
@@ -287,19 +287,18 @@ func (p *dockerNativeProvider) Diff(ctx context.Context, req *rpc.DiffRequest) (
 func diffUpdates(updates map[resource.PropertyKey]resource.ValueDiff) map[string]*rpc.PropertyDiff {
 	updateDiff := map[string]*rpc.PropertyDiff{}
 	for key, valueDiff := range updates {
-		if string(key) != "registry" {
-			updateDiff[string(key)] = &rpc.PropertyDiff{
-				Kind: rpc.PropertyDiff_UPDATE,
-			}
-		} else {
+		skipUpdate := false
+
+		if string(key) == "registry" && valueDiff.Object != nil {
```

Comment:
Unknown leads to Old and New properties field, because New is entirely unknown, so Object can't be built. See my test - it's the exact copy of what the issue repro leads to at runtime.

Reply:
Yes I understand the flow, I guess implicitly I'm asking are we interested in investing into representations where this class of issue cannot happen and is rejected at compile time (now probably easier with Go generics). Would that be something we can track and cross-link issues against?